File: Practica_3/src/Pila_Cola/Pila.java
package Pila_Cola;

public class Pila {
	private long[] arreglo;
	private int top;
	private int n;

	    // Constructor
	    public Pila(int n) {
	        this.n = n;
	        this.arreglo = new long[n];
	        this.top = -1;
	    }

	    // Agregar un Elemento
	    public void push(long e) {
	        if (isFull()) {
	            System.out.println("La pila está llena!");
	            return;
	        }
	        arreglo[++top] = e;
	    }

	    // Extraer un Elemento
	    public long pop() {
	        if (isEmpty()) {
	            System.out.println("La pila está vacía!");
	            return -1;
	        }
	        return arreglo[top--];
	    }

	    // Elemento en la Cima
	    public long peek() {
	        if (isEmpty()) {
	            System.out.println("La pila está vacía!");
	            return -1;
	        }
	        return arreglo[top];
	    }

	    // Verifica si la Pila esta vacia
	    public boolean isEmpty() {
	        return top == -1;
	    }

	    // Verifica si la Pila esta llena
	    public boolean isFull() {
	        return top == n - 1;
	    }
}

File: Practica_3/Pila.py
class Pila:
    def __init__(self, n):
        self.arreglo = [0] * n  #Arreglo de tamaño n
        self.top = -1  # Cima de la pila
        self.n = n  # Tamaño de la pila

    def push(self, e):
        if self.isFull():
            print("La pila esta llena!")
            return
        self.top += 1
        self.arreglo[self.top] = e

    def pop(self):
        if self.isEmpty():
            print("La pila esta vacia!")
            return None
        elemento = self.arreglo[self.top]
        self.top -= 1
        return elemento

    def peek(self):
        if self.isEmpty():
            print("La pila esta vacia!")
            return None
        return self.arreglo[self.top]

    def isEmpty(self):
        return self.top == -1

    def isFull(self):
        return self.top == self.n - 1

class Main:
   #Creando una Pila de 3 Elementos
    pila = Pila(3)
    #Adicionamos los 3 elementos
    print("Elementos de la Pila: ")
    pila.push(1)
    pila.push(2)
    pila.push(3)
    print(pila.arreglo[:pila.top + 1])
    #Ultimo elemento de la Pila Añadido
    print("Ultimo elemento:", pila.peek())
    #Extraemos un Elemento
    print("Elemento extraido:", pila.pop())
    #La Pila es Vacia
    print("¿La pila esta vacia?", pila.isEmpty())